src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -182,6 +183,11 @@ const NavBar: React.FC<NavBarPropsType> = ({ redirectionUrl }) => {
     []
   );
 
+  const playAudio = useMemo(() => {
+    const audioObj = new Audio(notificationSound);
+    numberOfUnreadMessages && audioObj.play();
+  }, [numberOfUnreadMessages]);
```

Comment:
This should probably use `useEffect` rather than `useMemo`. We also have no need to set a `const` from this, the `useEffect` will run and trigger the code automatically, based on the dependency (`numberOfUnreadMessages`) changing.

---

While I see that you're probably doing this check with the `&&` logic at the moment, it would be clearer if we used an `if` here, and checking that `numberOfUnreadMessages` is `> 0` before playing the audio.

We'll often use a 'guard' pattern of `if (!thingToBeChecked) return` in hooks like this, rather than wrapping the whole thing in a nested block.

Reply:
So this would end up looking something like:

```suggestion
  useEffect(() => {
    if (numberOfUnreadMessages <= 0) return
    
    new Audio(notificationSound).play();
  }, [numberOfUnreadMessages]);
```

Though this will create a new `Audio` object every time it needs to play, which i'm not sure if that's the best idea.